### PR TITLE
Product Tests w/ Implementations

### PR DESCRIPTION
**Pull request by poppabear8883**
This PR addressed coverage issues

### CODE REVIEW

File: tests/Unit/Products/ProductTest.php [hunk 1]
```diff
@@ -17,87 +17,79 @@ public function setUp(): void
         $this->repo = resolve(ProductRepositoryContract::class);
     }
 
+    private function createProduct($params = null): object
+    {
+        if ($params === null)
```
Comment (cbj4074) [inline tests/Unit/Products/ProductTest.php]: Didn't you already fix the missing parentheses? 😉

File: src/Base/Repositories/BaseRepository.php [hunk 2]
```diff
@@ -38,9 +38,5 @@ public function update(int $id, array $attributes): bool
         return $model->update($attributes);
     }
 
-    public function delete(int $id): bool
-    {
-        $model = $this->model()->find($id);
-        return $model->delete();
-    }
+    abstract public function delete(int $id): bool;
```
Comment (cbj4074) [inline src/Base/Repositories/BaseRepository.php:41]: What's the benefit of making this method abstract, vs. having a base method that would be sufficient for most implementations?

File: tests/Unit/Products/ProductTest.php (hunk 1 above)
Comment (poppabear8883) [reply tests/Unit/Products/ProductTest.php]: Crap

File: src/Base/Repositories/BaseRepository.php (hunk 2 above)
Comment (poppabear8883) [reply src/Base/Repositories/BaseRepository.php:41]: Read the body on the commit for this change

Comment (cbj4074) [reply src/Base/Repositories/BaseRepository.php:41]: Good point!

If and when more than one repository uses the exact same basic deletion logic, we should, in theory, be able to remove the `abstract` keyword and implement the base method without making any other changes, right?

Comment (poppabear8883) [reply src/Base/Repositories/BaseRepository.php:41]: Correct

**Review by cbj4074 — APPROVED** [review]

**Pull request merged**